text/0000-future-work.md

```diff
@@ -0,0 +1,171 @@
+[#2421]: https://github.com/rust-lang/rfcs/blob/master/text/2421-unreservations-2018.md#possible-future-unreservations
+[#2385]: https://github.com/Centril/rfcs/blob/rfc/implied-derive/text/0000-implied-derive.md#future-work
+[#2306]: https://github.com/rust-lang/rfcs/blob/master/text/2306-convert-id.md#possible-future-work
+
+The author of this RFC has benefitted personally from writing future-work
+sections ([#2532], [#2529], [#2524], [#2523], [#2522], [#2401], [#2421],
+[#2385], and [#2306]). Said written sections have also caused the current
+author to think more clearly about interactions in each of the written RFCs.
+If for no other reason, these sections offer a permantent space to idea-dump
+while writing an RFC.
+
+## For the team
+
+The holistic perspective that a future work section can offer may also
+help the relevant subteam to understand:
+
+1. why something is proposed,
+2. what the long term effects of said proposal is,
+4. how said proposals fit with the product vision and roadmap that the team
+   currently has.
+
+## For readers in general
+
+More generally, the benefits for the teams described above also hold for
+all readers. In particular, a reader can better infer what sort of language
+Rust is turning into given the information in a future-work section.
+Having such a section may also help generate interest in subsequent proposals
+which a different author may then write.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This Meta-RFC modifies the RFC template by adding a *"Future work"* section
+after the *"Unresolved questions"*. The newly introduced section is intended
+to help the authors, teams and readers in general reflect holistically on
+the big picture effects that a specific RFC proposal has.
+
+Please read the [reference-level-explanation] for exact details of what an
+RFC author will see in the changed template.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The implementation of this RFC consists of inserting the following text to the
+RFC template *after* the section *Unresolved questions*:
+
+> # Future work
+>
+> Think about what the natural extension and evolution of your proposal would
+> be and how it would affect the language and project as a whole in a holistic
+> way. Try to use this section as a tool to more fully consider all possible
+> interactions with the project and language in your proposal.
+> Also consider how the this all fits into the roadmap for the project
+> and of the relevant subteam.
+>
+> This is also a good place to "dump ideas", if they are out of scope for the
+> RFC you are writing but otherwise related.
+>
+> Note that having something written down in the future-work section is not
+> a reason to accept the current or a future RFC; such notes should be in the
+> section on motivation or rationale in this or subsequent RFCs.
+> The section merely provides additional information.
```

Comment:
I think this is pretty much exactly what I was thinking though I'd swap this and the next paragraph.

Ideally we'd indicate that this text is likely desirable to leave in the RFC (since it's both for authors and - perhaps even more so - for readers). I'm not sure how or if we should do that, though.

Reply:
Moved it around :)

My experience is that it isn't necessary (and it clutters the text) to keep the paragraph in the RFC since in the RFCs I wrote there was never too much unwarranted focus on the future work instead of what was actually proposed. However, if we notice it starts to become a problem, we can always change our minds later on and add a note to keep it.